code/main/consent/consentState.brs

```diff
@@ -0,0 +1,139 @@
+
+            if m._collectConsent = invalid
+                _adb_logVerbose("ConsentState::getCollectConsent() - Collect consent value not found in persistence. Extracting from configuration.")
+                m._collectConsent = m._getCollectConsentFromConfiguration()
+            end if
+
+            _adb_logVerbose("ConsentState::getCollectConsent() - Returning collect consent value: (" + FormatJson(m._collectConsent) + ")")
+            return m._collectConsent
+        end function
+
+        _loadCollectConsent: function() as dynamic
+            _adb_logVerbose("ConsentState::_loadCollectConsent() - Loading collect consent value from persistence.")
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            collectConsentValue = localDataStoreService.readValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.CONSENT_COLLECT)
+
+            return collectConsentValue
+        end function,
+
+        _saveCollectConsent: function(collectConsentValue as dynamic) as void
+            if not m._isValidConsentValue(collectConsentValue) then
+                _adb_logError("ConsentState::_saveCollectConsent() - Invalid collect consent value: (" + FormatJson(collectConsentValue) + ").")
+                return
+            end if
+
+            ''' cache the collect consent value
+            m._collectConsent = collectConsentValue
+
+            _adb_logVerbose("ConsentState::_saveCollectConsent() - Saving collect consent value " + FormatJson(collectConsentValue) + " in persistence.")
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            localDataStoreService.writeValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.CONSENT_COLLECT, collectConsentValue)
+        end function,
+
+        _deleteCollectConsent: function() as void
+            _adb_logVerbose("ConsentState::_deleteCollectConsent() - Deleting collect consent value from persistence.")
+
+            m._collectConsent = invalid
+
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            localDataStoreService.removeValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.CONSENT_COLLECT)
+            return
+        end function,
+
+        _getCollectConsentFromConfiguration: function() as dynamic
+            _adb_logVerbose("ConsentState::_getCollectConsentFromConfiguration() - Getting collect consent value from configuration.")
+            defaultConsentConfiguration = m._configurationModule.getDefaultConsent()
+            collectConsentValue = m._extractCollectConsentValueFromConfig(defaultConsentConfiguration)
+
+            return collectConsentValue
+        end function,
+
+        _isValidConsentValue: function(consentValue as dynamic) as boolean
+            if _adb_isEmptyOrInvalidString(consentValue) then
+                return false
+            end if
+
+            return (consentValue = "y" or consentValue = "p" or consentValue = "n")
+        end function,
```

Comment:
1) I think we should allow all consents and not restrict to just collect consent.
2) If we allow all consents, then the consent module doesn't need to validate the values as it doesn't use them. The Edge module does read the collect consent value so the code to parse the value should be put there. Also, in iOS/Android, the collect consent is parsed as "y", "n", and anything else is set to pending.
3) If you do really need this function, rename to something like `_isValidCollectConsentValue`

Reply:
It would be set by Edge module but this code is there just for the case when we want to retrieve collect consent from Configuration which needs parsing and validation. 

Only allowing collect consent because that is what is needed to make a decision to either send/queue/drop requests.

Reply:
This function is used when processing the Edge response handles. Most likely the collect consent value will be "y" or "n", but the standard does allow other values. If Konductor returns back a value other than "y", "n", or "p" (unlikely but possible) then the value is dropped because of this check.